catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/MetacardApplication.java

```diff
@@ -224,6 +224,11 @@ private String getSubjectEmail() {
     return SubjectUtils.getAttribute(SecurityUtils.getSubject(), Constants.ROLES_CLAIM_URI);
   }
 
+  // TODO: DDF-4249 to refactor this logic for PreQueryPlugin Access Control
+  private boolean isElevatedUser(List<String> subjectRoles) {
+    return subjectRoles.contains("system-user");
```

Comment:
❓Should this really be hardcoded? @codice/security

Reply:
It would be better to make this configurable. If a system was configured to use LDAP user store, users in that may not have this attribute, let alone the admin knowing they need this specific role.

Reply:
This is not an end goal solution, but a quick fix to address workspace performance issue. So i don't want to over engineering it. The real fix is captured in https://codice.atlassian.net/browse/DDF-4249. Also this fix is to align with https://github.com/codice/ddf/blob/5600fbd88f54292fb3ae84294c23cc47111bcc3a/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AccessControlSecurityConfiguration.java#L20, which is not configurable.

Reply:
I don't think this is a good quick fix. There's no guarantee any users will have this role

Reply:
❗️ This is not an acceptable quick fix.

Reply:
ahh i miss that. I'll make use of `AccessControlSecurityConfiguration` to get the 'systemUserAttributeValue'